.github/workflows/build.yml

```diff
@@ -17,135 +66,170 @@ jobs:
+          path: docs/_build/
+      - name: upload docs to github pages
+        if: github.event_name != 'pull_request' && github.ref == 'refs/heads/main' && runner.os == 'Linux'
+        uses: peaceiris/actions-gh-pages@de7ea6f8efb354206b205ef54722213d99067935 # pinned to v3.9.0
+        with:
+          github_token: ${{ secrets.GITHUB_TOKEN }}
+          publish_dir: ./docs/_build/html
+      - name: Create coverage report
+        run: nox -s coverage
+      - name: Upload comprehensive coverage HTML report
+        uses: actions/upload-artifact@v3
+        with:
+          name: coverage-report-${{ runner.os }}
+          path: htmlcov/
+      - name: Post coverage summary
+        shell: bash
+        run: |
+          echo "### ${{ runner.os }}"$'\n' >> $GITHUB_STEP_SUMMARY
+          cat .coverage_.md >> $GITHUB_STEP_SUMMARY
 
   compare-wheels:
     runs-on: ubuntu-latest
-    needs:
-      - build
+    needs: [build]
     steps:
-    - name: Download Linux package
-      uses: actions/download-artifact@v3
-      with:
-        name: python-packages-Linux
-        path: dist-Linux
-    - name: Download macOS package
-      uses: actions/download-artifact@v3
-      with:
-        name: python-packages-macOS
-        path: dist-macOS
-    - name: Download Windows package
-      uses: actions/download-artifact@v3
-      with:
-        name: python-packages-Windows
-        path: dist-Windows
-    - name: Unzip wheels
-      run: |
-        for os in Linux Windows macOS; do
-          unzip -d pkg-$os dist-$os/*.whl
-        done
-    - name: Compare macOS wheel to Linux wheel
-      run: diff -ur --color=always pkg-Linux pkg-macOS
-    - name: Compare Windows wheel to Linux wheel
-      run: diff -ur --color=always pkg-Windows pkg-Windows
+      - name: Download Linux package
+        uses: actions/download-artifact@v3
+        with:
+          name: python-packages-Linux
+          path: dist-Linux
+      - name: Download macOS package
+        uses: actions/download-artifact@v3
+        with:
+          name: python-packages-macOS
+          path: dist-macOS
+      - name: Download Windows package
+        uses: actions/download-artifact@v3
+        with:
+          name: python-packages-Windows
+          path: dist-Windows
+      - name: Unzip wheels
+        run: |
+          for os in Linux Windows macOS; do
+            unzip -d pkg-$os dist-$os/*.whl
+          done
+      - name: Compare macOS wheel to Linux wheel
+        run: diff -ur --color=always pkg-Linux pkg-macOS
+      - name: Compare Windows wheel to Linux wheel
+        run: diff -ur --color=always pkg-Windows pkg-Windows
 
   python-publish-package:
     # Only publish package on push to tag or default branch.
     if: ${{ github.event_name == 'push' && github.repository == 'jbms/sphinx-immaterial' && (startsWith(github.ref, 'refs/tags/v') || github.ref == 'refs/heads/main') }}
     runs-on: ubuntu-latest
-    needs:
-      - build
+    needs: [build, test]
```

Comment:
I wasn't sure if it was desired to deploy to pypi _after_ the docs and unit tests complete/pass. Because the pkg build is now done in an isolated job matrix, I figured I'd keep the original behavior in CI even though the build jobs complete much earlier than the tests jobs.

Reply:
I think it is probably better to keep the original behavior.  The idea was that we only tag a release after manually verifying that the build (on main branch) was successful.  Therefore it is not necessary to wait again for the tests to complete (or even run them at all).  Also there might be a case where a test is known to be failing for some reason but we still want to tag a release.

Reply:
ok, so I'll leave this as is then